terrors.go

```diff
@@ -22,6 +22,11 @@ var (
 		terrors.ErrPreconditionFailed: http.StatusPreconditionFailed,
 		terrors.ErrTimeout:            http.StatusGatewayTimeout,
 		terrors.ErrUnauthorized:       http.StatusUnauthorized,
+
+		// If the client disconnects it won't receive this response status code
+		// However, this allows intermediate proxies to distinguish this behaviour
+		// giving similar behaviour to nginx
+		terrors.ErrContextCanceled: 499,
```

Comment:
Perhaps `555` or something is more appropriate? If it's a bit of a mixed bag I would be more tempted to choose something in the 5XX range because for those where the client does is not disconnecting I think it would receive the error code? And 4XX means it's their fault which would then be incorrect.

Reply:
Ok, I'll see if we can pin down when it's a specific client disconnection causing the context cancellation.

Reply:
In the broader sense a context cancellation is effectively an interrupt, allowing a caller to signal to a callee that it should not waste effort producing a result which is no longer useful, but here afaik a context cancellation _does_ always mean that the client cancelled the request.